aip/0126.md

```diff
@@ -0,0 +1,102 @@
+---
+aip:
+  id: 126
+  state: reviewing
+  created: 2019-07-24
+permalink: /126
+redirect_from:
+  - /0126
+---
+
+# Enumerations
+
+It is common for a field to only accept or provide a discrete and limited set
+of values. In these cases, it can be useful to use enumerations (generally
+abbreviated "enums") in order to clearly communicate what the set of allowed
+values are.
+
+## Guidance
+
+APIs **may** expose enum objects for sets of values that are expected to change
+infrequently:
+
+```proto
+// A representation of a book.
+message Book {
+  // Other fields...
+
+  // Possible editions that may be published.
+  enum Edition {
+    // Default value. This value is unused.
+    EDITION_UNSPECIFIED = 0;
+
+    // The printed edition, in hardback.
+    HARDBACK = 1;
+
+    // The printed edition, in paperback.
+    PAPERBACK = 2;
+
+    // An e-book edition.
+    EBOOK = 3;
+
+    // An audio recording.
+    AUDIOBOOK = 4;
+  }
+
+  // The edition of the book.
+  Edition edition = 99;
+
+  // Other fields...
+}
+```
+
+- All enum values **must** use `SCREAMING_SNAKE_CASE`.
+- The first value of the enum **should** be the name of the enum itself
+  followed by the suffix `_UNSPECIFIED`.
+  - An exception to this rule is if there is a clearly useful zero value. In
+    particular, if an enum needs to present an `UNKNOWN`, it is usually clearer
+    and more useful for it to be a zero value rather than having both.
+- Enums which will only be used in a single message **should** be nested within
+  that message. In this case, the enum **should** be declared immediately
+  before it is used.
+
+### When to use enums
+
+Enums can be more accessible and readable than strings or booleans in many
+cases, but they do add overhead when they change. Therefore, enums **should**
+receive new values infrequently. While the definition of "infrequently" may
+change based on individual use cases, a good rule of thumb is no more than once
+a year. For enums that change frequently, the API **should** use a string and
+document the format.
+
+Additionally, enums **should not** be used when there is a competing,
+widely-adopted standard representation (such as with [language codes][bcp-47]
+or [media types][]).
```

Comment:
Or when the same values are likely to be used across multiple APIs, even if there isn't a *generally*-adopted standard representation?

Reply:
It would be okay to use _the same_ enum in this situation. (Speech and Dialogflow actually do this.) It would be worth being explicit about that though.

Reply:
Done.